### PR TITLE
Add io.github.OkuBrowser.oku

### PR DESCRIPTION
**Pull request by emmyoh**
### Please confirm your submission meets all the criteria

- [x] Please describe the application briefly. A Web browser that includes P2P capabilities (see [project site](https://okubrowser.github.io) for details). The `finish-args-flatpak-spawn-access` lint fails as the app needs FUSE access (for mounting replicas).
- [x] The domain used for the application ID is [controlled by the application developer(s)][appid-domain] and the [application id guidelines][appid] are followed.
- [x] I have read and followed all the [Submission requirements][reqs] and the [Submission guide][reqs2].
- [x] I have [built][build] and tested the submission locally.
- [x] I am an author/developer/upstream contributor to the project. If not, I contacted upstream developers about this submission. **Link:**

[appid-domain]: https://docs.flathub.org/docs/for-app-authors/requirements/#control-over-domain-or-repository
[appid]: https://docs.flathub.org/docs/for-app-authors/requirements#application-id
[reqs]: https://docs.flathub.org/docs/for-app-authors/requirements
[reqs2]: https://docs.flathub.org/docs/for-app-authors/submission
[build]: https://docs.flathub.org/docs/for-app-authors/submission/#before-submission

### CODE REVIEW

File: io.github.OkuBrowser.oku.json [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+        "--share=network",
+        "--share=ipc",
+        "--socket=fallback-x11",
+        "--socket=wayland",
+        "--device=all",
+        "--socket=pulseaudio",
+        "--socket=cups",
+        "--filesystem=xdg-download",
+        "--filesystem=xdg-pictures",
+        "--filesystem=host",
```
Comment (bbhtt) [inline io.github.OkuBrowser.oku.json:20]: you don't need the first 2 if you have host

File: io.github.OkuBrowser.oku.json [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+        "--filesystem=xdg-run/gvfs",
+        "--filesystem=xdg-run/gvfsd",
+        "--talk-name=org.freedesktop.Flatpak",
```
Comment (bbhtt) [inline io.github.OkuBrowser.oku.json:23]: this needs an exception PR https://docs.flathub.org/docs/for-app-authors/linter#exceptions https://docs.flathub.org/docs/for-app-authors/linter#finish-args-flatpak-spawn-access

File: io.github.OkuBrowser.oku.json [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+        {
+            "name": "host-command-wrapper",
+            "buildsystem": "simple",
+            "build-commands": [
+              "install fusermount-wrapper.sh /app/bin/fusermount3"
+            ],
+            "sources": [
+              {
+                "type": "file",
+                "path": "./fusermount-wrapper.sh"
```
Comment (bbhtt) [inline io.github.OkuBrowser.oku.json:96]: merge this with the module above.

Comment (bbhtt) [reply io.github.OkuBrowser.oku.json:96]: or with the libfuse module.

File: io.github.OkuBrowser.oku.json (hunk 1 above)
Comment (hfiguiere) [reply io.github.OkuBrowser.oku.json:20]: but then maybe you don't need host.

File: io.github.OkuBrowser.oku.json [hunk 4]
```diff
@@ -0,0 +1,101 @@
...
+                    "tag": "fuse-3.16.1",
+                    "commit": "1f0dfae4084577997291bb0e1b94aeff89a5e70f"
+                }
+            ]
+        },
+        {
+            "name" : "oku",
+            "buildsystem" : "simple",
+            "build-commands" : [
+                "./prebuild.sh",
```
Comment (hfiguiere) [inline io.github.OkuBrowser.oku.json:66]: what does that `prebuild.sh` do?

File: io.github.OkuBrowser.oku.json [hunk 5]
```diff
@@ -0,0 +1,101 @@
...
+                    "tag": "v0.1.0"
+                },
+                "cargo-sources.json"
+            ]
+        },
+        {
+            "name": "host-command-wrapper",
+            "buildsystem": "simple",
+            "build-commands": [
+              "install fusermount-wrapper.sh /app/bin/fusermount3"
```
Comment (hfiguiere) [inline io.github.OkuBrowser.oku.json:91]:
```suggestion
              "install -Dm755 fusermount-wrapper.sh /app/bin/fusermount3"
```

File: io.github.OkuBrowser.oku.json (hunk 4 above)
Comment (emmyoh) [reply io.github.OkuBrowser.oku.json:66]: For this version: https://github.com/OkuBrowser/oku/blob/v0.1.0/prebuild.sh

It compiles the `gresources` file. In the future, anything else that also needs running before compiling with Cargo will go in that same file.

File: io.github.OkuBrowser.oku.json (hunk 2 above)
Comment (emmyoh) [reply io.github.OkuBrowser.oku.json:21]: Exception PR: https://github.com/flathub-infra/flatpak-builder-lint/pull/538

File: io.github.OkuBrowser.oku.json (hunk 3 above)
Comment (emmyoh) [reply io.github.OkuBrowser.oku.json]: Done

File: io.github.OkuBrowser.oku.json (hunk 5 above)
Comment (emmyoh) [reply io.github.OkuBrowser.oku.json]: Done

File: io.github.OkuBrowser.oku.json (hunk 1 above)
Comment (emmyoh) [reply io.github.OkuBrowser.oku.json]: First two are redundant and have been removed. The browser needs full filesystem access.

Comment (bbhtt) [reply io.github.OkuBrowser.oku.json]: > The browser needs full filesystem access.

Why? Other browsers like Firefox does not need this.

**Comment by bbhtt** [comment]
> s as the app needs FUSE access (for mounting replicas).

Can you explain a bit more, what this does? This seems different than other browsers.

**Comment by emmyoh** [comment]
> > s as the app needs FUSE access (for mounting replicas).
> 
> 
> 
> Can you explain a bit more, what this does? This seems different than other browsers.

You'll find a more in-depth description on the site, but the gist of it is that Oku has 'replicas'—virtual drives that can be shared P2P—that can contain anything (including static sites) accessible in the browser. You may also mount these drives to either edit them (if you have the permissions to) or to interact with the contents outside the browser.

**Pull request closed without merging**